LEAF_Request_Portal/js/pages/mass_action.js

```diff
@@ -0,0 +1,352 @@
+ */
+function listRequests(queryObj, thisSearchID, getReminder = 0)
+{
+    $('#searchResults').hide();
+    $('#errorMessage').hide();
+    $('table#requests tr.requestRow').remove();
+    $('#iconBusy').show();
+
+    $.ajax({
+        type: 'GET',
+        url: './api/?a=form/query',
+        data: {q: JSON.stringify(queryObj),
+            CSRFToken: massActionToken},
+        cache: false
+    }).done(function(data) {
+        if(thisSearchID === searchID)
+        {
+            if(Object.keys(data).length)
+            {
+                $.each(data, function( index, value ) {
+                    let displayRecord = true;
+                    // If this is email reminder list, then compare against give time period
+                    if (getReminder) {
+                        // Get if we can show record for time period selected
+                        let numberActions = value.action_history.length;
+                        let lastActionDate = Number(value.action_history[numberActions-1].time) * 1000;
+
+                        // Current date minus selected reminder time period
+                        let comparisonDate = Date.now();
+                        comparisonDate -= (getReminder * 86400 * 1000);
+                        if (lastActionDate >= comparisonDate) {
+                            displayRecord = false;
+                        }
+                    }
+                    if (displayRecord) {
+                        requestsRow = '<tr class="requestRow">';
+                        requestsRow += '<td><a href="index.php?a=printview&amp;recordID=' + value.recordID + '">' + value.recordID + '</a></td>';
+                        requestsRow += '<td>' + ((value.categoryNames === undefined || value.categoryNames.length === 0) ? 'non' : value.categoryNames[0]) + '</td>';
+                        requestsRow += '<td>' + (value.service == null ? '' : value.service) + '</td>';
+                        requestsRow += '<td>' + value.title + '</td>';
+                        requestsRow += '<td><input type="checkbox" name="massActionRequest" class="massActionRequest" value="' + value.recordID + '"></td>';
+                        requestsRow += '</tr>';
+                        $('table#requests').append(requestsRow);
+                    }
+                });
+                $('#searchResults').show();
+            }
+            else
+            {
+                $('#errorMessage').html('No Results').show();
+            }
+        }
+    }).fail(function (jqXHR, error, errorThrown) {
+        console.log(jqXHR);
+        console.log(error);
+        console.log(errorThrown);
+    }).always(function (){
+        $('#iconBusy').hide();
+    });
+}
+
+/**
+ * Executes the selected action on each request selected in the table
+ */
+function executeMassAction()
+{
+    let selectedRequests = $('input.massActionRequest:checked');
+    let reminderDaysSince = Number($('#lastAction').val());
+    processedRequests = 0;
```

Comment:
Since I'm harassing you about stuff anyway, could you throw a comment in here about "Updating global variables"?  I was looking at this and was going "Ok, he scoped those other two variables, why didn't he these?"  And when I looked through the rest of the function and didn't see them.  It'll just help cut down on confusion later.

Reply:
Added a comment about them being global variables and where they were used elsewhere in the code to make sure others know where they are declared and used.